### ISSUE TITLE
kbfirmware.com warning: 'Not every key has a unique position in the wiring matrix.'

### ISSUE THREAD

**Issue by sohocoke** [LouWii/ErgoMax#13]
I'm in the process of mapping keys and ran into a problem.

I grabbed raw data from http://www.keyboard-layout-editor.com/#/gists/19051a469164c7ed1d6c813b2fb81ad2 and loaded it up in https://kbfirmware.com, then started changing some keys. I'm ready to test out the result, but can't download the firmware, which seems to be due to the following error:

`Not every key has a unique position in the wiring matrix.`

Any idea how I might fix this?

**Comment by LouWii** [comment]
The full layout on KLE wasn't meant to be used in kbfirmware. I'm not surprised it's not working.

Have you tried following the steps in the firmware doc instead? https://github.com/LouWii/ErgoMax/blob/master/Guide/FIRMWARE.md

**Comment by sohocoke** [comment]
@LouWii Since I built the reversible version, I thought I'd need to create one firmware that has the layout for both sides.

The steps in the firmware doc mention 2 separate layouts – does this mean I should flash left and right sides with the left and right layouts respectively?

If so, I will attempt to create the missing json file for the right side and try it.

**Comment by LouWii** [comment]
Yes, you need to flash each side separately with their respective firmware.
